### DISCUSSION TITLE
Question about instance update behaviour with JDBC data (documentation/javadoc)

### DISCUSSION THREAD

**Discussion by codejanovic** [micronaut-projects/micronaut-data#2010] [Q&A] [answered]
### Issue description

Hello fellow micronauts,

I just stumbled across the documentation on instance update with JDBC data as defined within the current docs [here](https://github.com/micronaut-projects/micronaut-data/blob/master/src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc#updating-an-instance-update), where it says:

>With Micronaut Data JDBC, **you must manually implement an update method** since the JDBC implementation doesn’t include >any dirty checking or persistence session notion. So you have to define explicit update methods for updates in your repository

This documentation seems to be me outdated, but I just want to make sure that I dont miss something. 

The javadoc on the other hand says the following (although I understand its the doc of a more abstract interface CrudRepository):
> This method issues an explicit update for the given entity. The method differs from {@link #save(Object)} in that an update will be generated regardless if the entity has been saved previously or not. If the entity has no assigned ID then an exception will be thrown.

## Current behaviour in my tests seems to be:

- calling the instance update method on a present entity...
  - does not throw any error
  - does update the changed fields accordingly
  - return the updated entity 
- calling the instance update method on an absent entity
  - does not throw any error
  - does not update anything
  - returns the entity (unchanged) passed into the instance update method call

## Questions:
- Is the documentation just outdated, or should users still not use the instance update method for any reasons?
- Is the javadoc accurate enough, as it implies a very different behaviour than the actual one?

**Comment by dstepanov** [comment, marked as answer]
> calling the instance update method on an absent entity 

This looks like an error. If the update returns zero rows changed, it should throw an error. IMHO, no error should be thrown if you define an update method with a number as a result. That means you will handle the update result yourself.

**Reply by codejanovic** [reply]
Thanks for the fast response! I agree that not throwing an error on the default instance method is propably an error. I can file an Bugissue later if you want me to. 

For the documentation part (online, not javadoc) I would guess, that it is simply outdated, right? If so, I could provide a Ticket/PR for that too (if desired)

**Reply by dstepanov** [reply]
Yeah, I think the documentation was trying to say that the partial update is impossible to do without dirty checking and the session, and the update needs to update all the fields.
PRs are very much welcome.

**Reply by codejanovic** [reply]
Alright, thanks for the clarification!
